assets/js/script.js

```diff
@@ -0,0 +1,990 @@
+        })
+      }
+      if (this.spdY < 0 && this.spdY > this.spdX) {
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 3) * 70}px -20px`
+        })
+      }
+      this.aniFrame += 0.07
+      //enemy collision with border
+      if (this.x <= 0 || this.x >= gameWidth - this.sizeX) this.spdX *= -1 //border collision x
+      if (this.y <= 0 || this.y >= gameHeight - this.sizeY) this.spdY *= -1 //border collision y
+    }
+
+    if (this.type === "upgrade") {
+      this.x += this.spdX
+      this.y += this.spdY
+      if (this.x <= 0 || this.x >= gameWidth - this.sizeX) this.spdX *= -1 //border collision x
+      if (this.y <= 0 || this.y >= gameHeight - this.sizeY) this.spdY *= -1 //border collision
+    }
+
+    if (this.type === "cat") {
+      this.x += this.spdX
+      this.y += this.spdY
+      if (this.spdX === 0) {
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 4) *
+            -67.5}px -270px`
+        })
+      }
+      if (this.spdX > 0 && this.spdY > 0) {
+        //right
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 4) *
+            -67.5}px -200px`
+        })
+      }
+      if (this.spdX < 0 && this.spdY > 0) {
+        //left
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 4) * -67.5}px 0px`
+        })
+      }
+      if (this.spdX > 0 && this.spdY < 0) {
+        //down
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 4) *
+            -67.5}px -132px`
+        })
+      }
+      if (this.spdX < 0 && this.spdY < 0) {
+        //up
+        this.jTarget.css({
+          backgroundPosition: `${Math.floor(this.aniFrame % 4) * -67.5}px -65px`
+        })
+      }
+      this.aniFrame += 0.1
+
+      //enemy collision with border
+      if (this.x <= 0 || this.x >= gameWidth - this.sizeX) this.spdX *= -1 //border collision x
+      if (this.y <= 0 || this.y >= gameHeight - this.sizeY) this.spdY *= -1 //border collision y
+    }
+
+    if (this.type === "bullet") {
+      this.x += this.spdX
+      this.y += this.spdY
+      if (this.x <= 0 || this.x >= gameWidth - this.sizeX) this.removeChar() //border collision x
+      if (this.y <= 0 || this.y >= gameHeight - this.sizeY) this.removeChar()
+      if (Math.floor(this.aniFrame % 4) === 0) {
+        this.jTarget.css({
+          backgroundPosition: `0 0`,
+          transform: `rotate(${this.aimAngle - 90}deg)`
+        })
+      } else if (Math.floor(this.aniFrame % 4) === 1) {
+        this.jTarget.css({ backgroundPosition: `50px 0` })
+      } else if (Math.floor(this.aniFrame % 4) === 2) {
+        this.jTarget.css({ backgroundPosition: `0 -48px` })
+      } else {
+        this.jTarget.css({ backgroundPosition: `50px -48px` })
+      }
+      this.aniFrame += 0.1
+    }
+
+    if (this.type === "player") {
+      $body.on("keydown", e => {
```

Comment:
If I suggest moving those eventListners to the outside of Class method `moveChar()`, where you want to put that?